app/src/main/java/protect/card_locker/Utils.java

```diff
@@ -163,9 +161,20 @@ static public BarcodeValues parseSetBarcodeActivityResult(int requestCode, int r
         throw new UnsupportedOperationException("Unknown request code for parseSetBarcodeActivityResult");
     }
 
+    static public Bitmap retrieveImageFromUri(Context context, Uri data) throws IOException {
+        Bitmap bitmap;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            ImageDecoder.Source image_source = ImageDecoder.createSource(context.getContentResolver(), data);
+            bitmap = ImageDecoder.decodeBitmap(image_source, (decoder, info, source) -> decoder.setMutableRequired(true));
+        } else {
+            bitmap = getBitmapSdkLessThan29(data, context);
+        }
+        return bitmap;
+    }
```

Comment:
I think this can just return straight away:

```suggestion
    static public Bitmap retrieveImageFromUri(Context context, Uri data) throws IOException {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            ImageDecoder.Source image_source = ImageDecoder.createSource(context.getContentResolver(), data);
            return ImageDecoder.decodeBitmap(image_source, (decoder, info, source) -> decoder.setMutableRequired(true));
        } else {
            return getBitmapSdkLessThan29(data, context);
        }
    }
```